### PR TITLE
[terra-framework] Update terra-form-select dependency

### LINKED ISSUE #1151: Update terra-form-select dependency in framework packages
# Feature Request
Update terra-form-select dependency in framework packages

## Description
Within terra-framework 3 packages have dependencies on v5 of terra-form-select:
terra-dialog-modal
terra-form-validation
terra-modal-manager

## Additional Context / Screenshots
terra-form-select is currently on v6.6.0

### PR DESCRIPTION
**Pull request by pranav300**
### Summary

This PR updates the packages 
* terra-dialog-modal 
* terra-form-validation
* terra-modal-manager

To consume the latest major version of `terra-form-select` i.e. `v6.x.y`.

Updated Jest snapshots failing due to the changes of terra-icon : https://github.com/cerner/terra-core/pull/3023/files.

Closes #1151 

### Deployment Link

https://terra-framework-deployed-pr-1154.herokuapp.com/!--

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-dialog-modal/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ Changelog
 
 Unreleased
 ----------
+### Changed
+* Consume latest major version bump of terra-avatar
```
Comment (supreethmr) [inline packages/terra-dialog-modal/CHANGELOG.md:7]:
```suggestion
* Updated terra-form-select to V6
```

Comment (pranav300) [reply packages/terra-dialog-modal/CHANGELOG.md]: updated here https://github.com/cerner/terra-framework/pull/1154/commits/df271ac7635f575ce13f0935b0630ea2603a15ad

File: packages/terra-date-picker/tests/jest/__snapshots__/DateInput.test.jsx.snap [hunk 2]
```diff
@@ -153,7 +153,7 @@ exports[`correctly applies the theme context className 1`] = `
                   >
                     <svg
                       aria-hidden="true"
-                      className="tui-Icon icon icon-svg"
+                      className="tui-Icon icon orion-fusion-theme icon-svg"
```
Comment (benbcai) [inline packages/terra-date-picker/tests/jest/__snapshots__/DateInput.test.jsx.snap:156]: These changes (including the terra-menu screenshot update) are already being added in https://github.com/cerner/terra-framework/pull/1153.

File: packages/terra-modal-manager/package.json [hunk 3]
```diff
@@ -37,7 +37,7 @@
   },
   "devDependencies": {
     "terra-collapsible-menu-view": "^6.35.0",
-    "terra-form-select": "^5.0.0"
+    "terra-form-select": "^6.8.0"
```
Comment (emilyrohrbough) [inline packages/terra-modal-manager/package.json:40]: Where is this dependency being used?

Comment (pranav300) [reply packages/terra-modal-manager/package.json:40]: It is used in the test [DisclosureIntegrationComponent.jsx](https://github.com/cerner/terra-framework/blob/master/packages/terra-modal-manager/src/terra-dev-site/test/modal-manager/DisclosureIntegrationComponent.jsx#L4)

Comment (emilyrohrbough) [reply packages/terra-modal-manager/package.json:40]: Figured it was but missed it when I checked

**Pull request merged**